### DISCUSSION TITLE
#<NameError: uninitialized constant GraphQL::Pro::PunditIntegration::HasPunditRole::Pundit>

### DISCUSSION THREAD

**Discussion by sujitrai05** [rmosolgo/graphql-ruby#4011] [Q&A]
This is how I am setting up the resolver field.
field :subscription, Types::Subscription::InstanceType, null: true, default_value: {}, pundit_role: :subscription

Result is coming as expected, however, I get error "#<NameError: uninitialized constant GraphQL::Pro::PunditIntegration::HasPunditRole::Pundit>"

I have included all the pundit field dependencies on base classes.

Why am i getting this issue? What I might have done wrong or wrongly configured?

**Comment by rmosolgo** [comment]
Did you add pundit to your project? If not, try `$ bundle add pundit` (or `$ gem install pundit`, then add `require "pundit"` to your app). Sorry that assumption isn't well documented! If it works for you, I'll update the code to raise a helpful error when that constant is missing, let me know what you find.

**Reply by sujitrai05** [reply]
yes it does.

**Reply by sujitrai05** [reply]
@rmosolgo, after setting up the pundit, I am getting new runtime error now. I do get my expected result but there's this run time error too.

#<GraphQL::Pro::PunditIntegration::PolicyNotFoundError: No policy found for:

- Type: Query
- Runtime value: `nil`

Pundit error: Pundit::NotDefinedError, unable to find policy `NilClassPolicy` for `nil`


This is expected behaviour?

**Reply by rmosolgo** [reply]
Yes, that's the expected behavior. At runtime, by default, the Pundit integration checks the current `object` and tries to find a policy for it using Pundit's built-in lookup method, `Pundit.policy!(...)`. If it can't find a policy, it raises this error. 

Your options are: 

- Define the policy that Pundit expected to find. In this case, it was `NilClassPolicy` -- it doesn't really make sense to define that, so this option doesn't apply here. 
- Manually configure a policy class to use for this GraphQL object (`Query`), using `pundit_policy_class(...)`, as described here: https://graphql-ruby.org/authorization/pundit_integration.html#custom-policy-class. If you want to authorize the fields on your root query object, this might be a good option -- you could define a `QueryPolicy` and configure it using that method. 
- Skip policy checking by calling `pundit_policy_class(nil)` in the type definition. If you don't require any special permissions for any of your root fields, this would be a good option. (And, as a side note, if you want that for a default behavior, you can configure it in your base object type and all subclasses will inherit it.) 


I hope that helps! If you try one of those and it doesn't work for you, please share the code you tried and what happened when you tried it.
